packages/notifications/src/styled/StyledAlert.ts

```diff
@@ -5,37 +5,157 @@
+      pseudoColor = palette[dangerHue][800];
+      bgColor = getColor(dangerHue, 600, theme, 0.15);
+      break;
+    case 'warning':
+      color = palette[warningHue][700];
+      pseudoColor = palette[warningHue][800];
+      bgColor = getColor(warningHue, 700, theme, 0.15);
+      break;
+    case 'info':
+      color = palette[neutralHue][600];
+      pseudoColor = palette[neutralHue][800];
+      bgColor = getColor(neutralHue, 600, theme, 0.15);
+      break;
+    default:
+      color = 'inherit';
+      pseudoColor = 'inherit';
+      bgColor = 'inherit';
+      break;
+  }
+
+  return css`
+    ${StyledClose} {
+      color: ${color};
+    }
+
+    ${StyledClose}:focus {
+      outline: none;
+      background-color: ${bgColor};
+      color: ${pseudoColor};
+    }
+
+    ${StyledClose}:hover {
+      color: ${pseudoColor};
+    }
+  `;
+};
+
+const colorStyles = (props: IStyledAlertProps & ThemeProps<DefaultTheme>) => {
+  const { type, theme } = props;
+  const { colors, palette } = theme;
+  const { successHue, dangerHue, warningHue, neutralHue } = colors;
+
+  let color;
+  let titleColor;
+  let borderColor;
+  let backgroundColor;
+
+  switch (type) {
+    case 'success':
+      color = palette[successHue][700];
+      titleColor = palette[successHue][800];
+      borderColor = palette[successHue][300];
+      backgroundColor = palette[successHue][100];
+      break;
+    case 'error':
+      color = palette[dangerHue][700];
+      titleColor = palette[dangerHue][800];
+      borderColor = palette[dangerHue][300];
+      backgroundColor = palette[dangerHue][100];
+      break;
+    case 'warning':
+      color = palette[warningHue][700];
+      titleColor = palette[warningHue][800];
+      borderColor = palette[warningHue][300];
+      backgroundColor = palette[warningHue][100];
+      break;
+    case 'info':
+      color = palette[neutralHue][600];
+      titleColor = palette[neutralHue][800];
+      borderColor = palette[neutralHue][300];
+      backgroundColor = palette[neutralHue][100];
+      break;
+    default:
+      color = palette[neutralHue][600];
+      titleColor = palette[neutralHue][800];
+      borderColor = palette[neutralHue][300];
+      backgroundColor = colors.background;
+      break;
+  }
+
+  return css`
+    border-color: ${borderColor};
+    background-color: ${backgroundColor};
+    color: ${color};
+
+    ${StyledTitle} {
+      color: ${titleColor};
+    }
+  `;
+};
+
+const padding = (props: ThemeProps<DefaultTheme>) => {
+  const { space, rtl } = props.theme;
+  const paddingTop = space.md;
+  const paddingRight = rtl ? `${space.base * 12.5}px` : space.xl;
+  const paddingBottom = space.md;
+  const paddingLeft = rtl ? space.xl : `${space.base * 12.5}px`;
+
+  return `${paddingTop} ${paddingRight} ${paddingBottom} ${paddingLeft}`;
+};
+
 /**
  * Supports all `<div>` props
  */
-export const StyledAlert = styled.div.attrs<IStyledAlertProps>(props => ({
+export const StyledAlert = styled.div.attrs({
```

Comment:
> I had really hoped to see ideas around a generic StyledDiv fleshed out 

Please don’t lose hope yet 🙂. I’m all ears and would love to discuss and flesh out ideas around a generic `StyledDiv` now. We had briefly talked about this two days ago but I wasn’t 100% clear on what the generic `StyledDiv` meant and should have asked about an example.

I’ve continued to think about it. There seem to be five style properties, `position`, `border`, `border-radius`, `line-height`, and `font-size`, that are shared across the `<Alert>`, `<Notification>` and `<Well>` components which can be abstracted out into a [`<StyledDiv>`](https://github.com/zendeskgarden/react-components/blob/hzhu/notifications-css-in-js-ii/packages/notifications/src/styled/StyledDiv.ts). Then, the `<StyledDiv>` can be used to implement each of the notifications, like [this](https://github.com/zendeskgarden/react-components/blob/hzhu/notifications-css-in-js-ii/packages/notifications/src/styled/StyledAlert.ts#L142), [this](https://github.com/zendeskgarden/react-components/blob/hzhu/notifications-css-in-js-ii/packages/notifications/src/styled/StyledNotification.ts#L101), and finally [this](https://github.com/zendeskgarden/react-components/blob/hzhu/notifications-css-in-js-ii/packages/notifications/src/styled/StyledWell.ts#L33). Does this align with what you had in mind?

Reply:
Spoke with @jzempel offline about what a `<StyledDiv>` might look like. I'm in the process of getting rid of `<StyledAlert>`, `<StyledNotification>`, and `<StyledWell`> in favor of a `<StyledDiv>`. I will push up a new commit for this once it's fleshed out. Stay tuned.